zeebe/exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/ElasticsearchExporterIT.java

```diff
@@ -195,4 +210,165 @@ final var record = factory.generateRecord();
+
+      /* Tests when retention is later enabled all indices should have lifecycle policy */
+      // given
+      configureExporter(true);
+      final var record2 = factory.generateRecord(ValueType.JOB);
+
+      // when
+      export(record2);
+
+      // then
+      final var index2 = indexRouter.indexFor(record2);
+      final var response2 = testClient.getIndexSettings(index2);
+      assertIndexSettingsHasLifecyclePolicy(response2);
+
+      response1 = testClient.getIndexSettings(index1);
+      assertIndexSettingsHasLifecyclePolicy(response1);
+    }
+
+    @Test
+    @Order(2)
+    void shouldRemoveIndexLifecycleSettingsFromExistingIndicesOnRerunWhenRetentionIsDisabled() {
+      // given
+      configureExporter(true);
+      final var record1 = factory.generateRecord(ValueType.JOB);
+
+      // when
+      export(record1);
+
+      // then
+      final var index1 = indexRouter.indexFor(record1);
+      var response1 = testClient.getIndexSettings(index1);
+      assertIndexSettingsHasLifecyclePolicy(response1);
+
+      /* Tests when retention is later disabled all indices should not have a lifecycle policy */
+      // given
+      configureExporter(false);
+      final var record2 = factory.generateRecord(ValueType.JOB);
+
+      // when
+      export(record2);
+
+      // then
+      final var index2 = indexRouter.indexFor(record2);
+      final var response2 = testClient.getIndexSettings(index2);
+      assertIndexSettingsHasNoLifecyclePolicy(response2);
+
+      response1 = testClient.getIndexSettings(index1);
+      assertIndexSettingsHasNoLifecyclePolicy(response1);
+    }
+
+    /**
+     * Default timeout for elasticsearch `PUT /<target>/_settings` is 30 seconds.
+     *
+     * <p>500 records each has a shard and a replica means 1000 shards, which is the maximum open
+     * shards in a one node cluster
+     */
+    @Test
+    @Order(3)
+    void shouldNotTimeoutWhenUpdatingLifecyclePolicyForExistingIndices() {
+      // given
+      configureExporter(false);
+      final var records = new ArrayList<Record<RecordValue>>();
+      final int limit = 498 - exportedRecordsCounter;
+      for (int i = 0; i < limit; i++) {
+        records.add(factory.generateRecord(ValueType.JOB));
+      }
+
+      // when
+      for (final var record : records) {
+        export(record);
+      }
+
+      // then
+      for (final var record : records) {
+        final var index = indexRouter.indexFor(record);
+        final var response = testClient.getIndexSettings(index);
+
+        assertIndexSettingsHasNoLifecyclePolicy(response);
+      }
+
+      /* Tests when retention is later enabled all indices should have lifecycle policy */
+      // given
+      configureExporter(true);
+      final var record2 = factory.generateRecord(ValueType.JOB);
+
+      // when
+      await("New record is exported, and existing indices are updated")
+          .atMost(Duration.ofSeconds(30))
+          .until(() -> export(record2));
```

Comment:
🔧 I don't think we need to test the non-application of the policy here?  this is already covered by `shouldAddIndexLifecycleSettingsToExistingIndicesOnRerunWhenRetentionIsEnabled`, would thus suggest to shorten it a bit.

```suggestion
      // given
      configureExporter(false);
      final var records = new ArrayList<Record<RecordValue>>();
     // using 498 here as we will export one more record after (1 main shard, 1 replica)
      final int limit = 498;
      for (int i = 0; i < limit; i++) {
        final var record = factory.generateRecord(ValueType.JOB);
        records.add(record);
        export(record);
      }

      // when
      configureExporter(true);
      final var record2 = factory.generateRecord(ValueType.JOB);

      // when
      await("New record is exported, and existing indices are updated")
          .atMost(Duration.ofSeconds(30))
          .until(() -> export(record2));
```